lib/js/jslib.js

```diff
@@ -787,32 +787,39 @@ const LINKS = new (function() {
    * @param {any} state
    * @param {any} obj
    */
-  function resolveServerValue (state, obj) {
-    if (obj instanceof Object) {
-      if (obj.type) {
-        // XML, skip
-        return;
-      }
-      for (let i in obj) {
-        if(LINKEDLIST.isNil(obj[i]))
+  function resolveServerValue (state, root) {
+    let queue = [root];
+    while (queue.length > 0) {
+      let obj = queue.shift();
+
+      if (obj instanceof Object) {
+        if (obj.type) {
+          // XML, skip
           continue;
+        }
 
-         resolveServerValue(state, obj[i]);
+        for (let i in obj) {
+          if(LINKEDLIST.isNil(obj[i]))
+            continue;
 
-         if (obj[i].func) {
-           const f = (!TYPES.isObject(obj[i].environment)) ?
-             eval(obj[i].func) :
-             partialApply(eval(obj[i].func), eval(obj[i].environment));
-           f.location = obj[i].location; // This may be set to 'server' by the server serializer.
-           f.func = obj[i].func;
+          // Queue child object
+          queue.unshift(obj[i]);
 
-          obj[i] = f;
-        } else if (obj[i].key) {
-          obj[i].key = _lookupMobileKey(state, obj[i].key)
+          // Resolve this object
+          if (obj[i].func) {
+            const f = (!TYPES.isObject(obj[i].environment)) ?
+              eval(obj[i].func) :
```

Comment:
I have always been curious about this particular code fragment. Since closure conversion, I would expect every function to be available at the top-level. So I would believe something less powerful than `eval` ought to do it, say: `window[function_name](arguments)`.